src/ui/Base/Initialization.ts

```diff
@@ -242,41 +242,68 @@ export class Initialization {
     const alreadyInitialized = Component.get(element, QueryController, true);
     if (alreadyInitialized) {
       this.logger.error('This DOM element has already been initialized as a search interface, skipping initialization', element);
-      return new Promise((resolve, reject) => {
-        resolve({ elem: element });
+      return Promise.resolve({
+        elem: element
       });
     }
 
     options = Initialization.resolveDefaultOptions(element, options);
 
-    Initialization.performInitFunctionsOption(options, InitializationEvents.beforeInitialization);
+    const waitForAllPromisesToFinish = async (eventType: string, promises: Promise<any>[]) => {
+      const promisesWithErrorsHandledIndividually = promises.map(p => {
+        return p.catch(error =>
+          this.logger.warn(
+            `An error occurred when trying to defer the \"${eventType}\" event. The defer will be ignored.`,
+            `Error: ${error}`
+          )
+        );
+      });
+      return Promise.all(promisesWithErrorsHandledIndividually).catch(error =>
+        this.logger.error(
+          `An unexpected error occurred when trying to defer the \"${event}\" event. All defers will be ignored.`,
+          `Error: ${error}`
+        )
+      );
+    };
+
+    const triggerInitializationEventWithArguments = async (eventType: string) => {
+      const initializationEventArgs: IInitializationEventArgs = {
+        defer: []
+      };
+      $$(element).trigger(eventType, initializationEventArgs);
+      if (initializationEventArgs.defer.length > 0) {
```

Comment:
One of the changes required for the synchronous-ity of the non-lazy init.

The other being the `async` that I originally put on the method.